keps/sig-cluster-lifecycle/kubeadm/20191229-kubeadm-move-out-of-kk.md

```diff
@@ -0,0 +1,703 @@
+`v2.0.0`, the dual-tags (SEMVER) will overlap with the non-SEMVER tags.
+
+There are no good solutions for this problem and users might have to
+use SHAs to import the kubeadm module.
+
+See the related discussions at:
+https://github.com/kubernetes/kubernetes/issues/72638
+https://github.com/kubernetes/kubernetes/issues/84372
+
+## Design Details
+
+### Enhancements proposals
+
+The current plan is for kubeadm to continue following the k/enhancements
+and KEP process, given the project will be part of the k8s release.
+There will be no extra burden on the release team as they are already
+following PRs and issues in repositories other than k/k.
+
+There is also the option to decouple the enhancement process and have future
+kubeadm KEPs in the k/kubeadm repository, but needs further discussion
+and there are no apparent benefits.
+
+### Documentation
+
+#### Authored content
+
+There are no planned changes for how kubeadm will release its authored
+documentation. Kubeadm already has multiple documents in the k8s.io website
+and those are maintained with the help of sig-docs.
+
+#### Reference documentation
+
+The reference documentation update process needs changes.
+Currently tooling maintained by sig-docs does the following:
+1) Clones k/k
+2) Runs tools to generate cmd/kubeadm/app Cobra reference documentation
+3) Prettyfies the documentation with HTML/CSS
+
+The tooling needs to change to start generating the documentation
+from the k/kubeadm repository. SIG docs must be notified of this
+and collaborate with the kubeadm maintainers when the move is about to happen.
+
+There would not be a need for the tooling to continue building old
+reference documentation from k/k instead, as the k8s.io website reference
+documentation is only generated once before a release happens.
+
+The tools for building k/kubeadm reference docs in k/k must be removed.
+The initial proposal is to not support building reference documentation
+from the tools in the k/kubeadm repository, but might need some discussion.
+Currently the consumers of k/k reference documentation are unknown and the
+documentation is no longer tracked in commits. See:
+https://github.com/kubernetes/kubernetes/tree/master/docs
+
+#### Release notes
+
+The prow “release-note” plugin must be enabled for the kubeadm repository.
+This will make PRs require release notes or the option NONE. Once a k8s
+release has begun ideally the tool:
+https://github.com/kubernetes/release/tree/master/cmd/release-notes
+should be levereged to obtain all the release notes for kubeadm in this
+release.
+
+The alternative is for the kubeadm maintainers to collect the release
+notes manually and send them to the release team, but this might be more
+difficult to handle as the k8s release tooling sends emails with the release
+notes right away. Also the kubeadm maintainers might make mistakes forgetting
+about a certain feature/bug fix.
+
+### Test Plan
```

Comment:
This talks about the kubeadm repo but not how the kubernetes repo will handle this transition

Reply:
the kubernetes repository will exclude the kubeadm source code and the overall idea from SIG Arch is that k/k should not know about kubeadm. this might not be avoidable because kubeadm will still be part of the release.

we have a couple of options about building kubeadm for CI and release:
- build kubeadm using Anago. this is becoming a less favorable option at this point, even if it's in the main proposal.
- build kubeadm from k/k tooling, from `make release`, placing the binary in the same location where other artifacts are - kubelet, kubectl etc.

this is already covered in other sections.

do you have something else in mind?